client/trino-cli/src/main/java/io/trino/cli/QueryRunner.java

```diff
@@ -59,70 +40,27 @@ public class QueryRunner
     private final Consumer<OkHttpClient.Builder> sslSetup;
 
     public QueryRunner(
+            TrinoUri uri,
             ClientSession session,
             boolean debug,
-            HttpLoggingInterceptor.Level networkLogging,
-            Optional<HostAndPort> socksProxy,
-            Optional<HostAndPort> httpProxy,
-            Optional<String> keystorePath,
-            Optional<String> keystorePassword,
-            Optional<String> keystoreType,
-            Optional<String> truststorePath,
-            Optional<String> truststorePassword,
-            Optional<String> truststoreType,
-            boolean useSystemTruststore,
-            boolean insecureSsl,
-            Optional<String> accessToken,
-            Optional<String> user,
-            Optional<String> password,
-            Optional<String> kerberosPrincipal,
-            Optional<String> krb5ServicePrincipalPattern,
-            Optional<String> kerberosRemoteServiceName,
-            Optional<String> kerberosConfigPath,
-            Optional<String> kerberosKeytabPath,
-            Optional<String> kerberosCredentialCachePath,
-            boolean kerberosUseCanonicalHostname,
-            boolean delegatedKerberos,
-            boolean externalAuthentication,
-            List<ExternalRedirectStrategy> externalRedirectHandlers)
+            HttpLoggingInterceptor.Level networkLogging)
     {
         this.session = new AtomicReference<>(requireNonNull(session, "session is null"));
         this.debug = debug;
 
-        if (insecureSsl) {
-            this.sslSetup = OkHttpUtil::setupInsecureSsl;
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        try {
+            sslSetup = uri.getSetupSsl();
```

Comment:
why do we need this sslSetup at all?

Isn't `uri.setupClient(builder);` already doing required SSL initialization?